### PR TITLE
Clear out plan ref when plan changed using k8s names

### LINKED ISSUE #1552: Cannot change plan of a ServiceInstance using K8S names
The ServiceInstance strategy only clears `ClusterServicePlanRef` when `ClusterServicePlanExternalName` changes. It should also clear `ClusterServicePlanRef` when `ClusterServicePlanName` changes.

### PR DESCRIPTION
**Pull request by staebler**
Fixes #1552.

I have an integration test coming for this as part of my next round of integration test improvements.

### CODE REVIEW

**Comment by n3wscott** [comment]
LGTM

File: pkg/registry/servicecatalog/instance/strategy.go [hunk 1]
```diff
@@ -156,7 +156,8 @@ func (instanceRESTStrategy) PrepareForUpdate(ctx genericapirequest.Context, new,
 	newServiceInstance.Spec.ClusterServicePlanRef = oldServiceInstance.Spec.ClusterServicePlanRef
 
 	// Clear out the ClusterServicePlanRef so that it is resolved during reconciliation
-	if newServiceInstance.Spec.ClusterServicePlanExternalName != oldServiceInstance.Spec.ClusterServicePlanExternalName {
+	if newServiceInstance.Spec.ClusterServicePlanExternalName != oldServiceInstance.Spec.ClusterServicePlanExternalName ||
+		newServiceInstance.Spec.ClusterServicePlanName != oldServiceInstance.Spec.ClusterServicePlanName {
```
Comment (n3wscott) [inline pkg/registry/servicecatalog/instance/strategy.go]: It feels like we should ignore the ExternalPlanName at first thought. Why do we have to care they match?

Comment (staebler) [reply pkg/registry/servicecatalog/instance/strategy.go]: The user can specify the class and plan using *either* the external names or the k8s names.

If the plan names do not match, that indicates that the user has requested a change to the plan. In that case, the plan that the ServiceInstance referenced is no longer the plan that the ServiceInstance should reference. The API Server cannot resolve the reference, so the API Server clears out the reference. The Controller then attempts to resolve the reference during reconciliation of the spec changes made to the ServiceInstance.

**Comment by arschles** [comment]
@staebler will merge pending CI

**Comment by kibbles-n-bytes** [comment]
CI has passed, merging!

**Pull request merged**